cpp/src/arrow/csv/parser_test.cc

```diff
@@ -621,6 +645,25 @@ TEST(BlockParser, MismatchingNumColumns) {
     EXPECT_RAISES_WITH_MESSAGE_THAT(
         Invalid, testing::HasSubstr("CSV parse error: Expected 2 columns, got 1: a"), st);
   }
+  // Vary the number of columns and mismatch, to catch buffer overflow issues
+  for (int32_t num_cols : {1, 2, 5, 10, 100}) {
```

Comment:
Why do we need `2, 5, 10` here? `1, 100` aren't enough?

Reply:
The conditions for triggering the issue are bit mysterious, so I thought it would be safer to ensure non-regression with a larger set of parameters. I can trim it down a bit of course.